pkg/amber/endorsement.go

```diff
@@ -28,6 +30,23 @@ import (
 // version in `schema/amber-endorsement/v1`.
 const AmberEndorsementV2 = "https://github.com/project-oak/transparent-release/endorsement/v2"
 
+// VerifiedProvenanceSet encapsulates metadata about a non-empty list of verified provenances.
+type VerifiedProvenanceSet struct {
+	// Name of the binary that all validated provenances agree on.
+	BinaryName string
```

Comment:
I didn't realise that the binary name has any significance; is it something that needs to be checked explicitly?

Reply:
+1.

Reply:
I am not entirely sure if it is significant. If it has no significance, why is it included in the subject? This may be a question for in-toto actually.

If there are multiple provenances and they have different names, what should we do? In particular, which name should we use as the name in the subject of the generated endorsement? 

Finally, currently, in the Amber provenances, we include the Git commit hash in the name. This is a legacy decision, which will not be true for SLSA-generated provenances. So, perhaps the condition could be relaxed. 

Created a tracking issue: #144.